### PR TITLE
Added notification item code

### LINKED ISSUE #505: NEAR Social Notifications link to raw Wiki/Blog post types instead of the formatted versions within the blog widget
**Describe the bug**
Since introducing the blog post type, I noticed that they showed up in my near.social notifications. However, when I click on them – they point me to the [raw post](https://near.social/devgovgigs.near/widget/gigs-board.pages.Post?id=2029) instead of to the [formatted versions](https://near.social/devgovgigs.near/widget/app?page=blog&id=2029) within the blog widget.

**To Reproduce**
1. Go to https://near.social/mob.near/widget/NotificationFeed
2. Click on any notification related to editing a blog post type (you may need to edit one first)
3. Click View DevHub post
4. See raw version.

**Expected behavior**
Ideally we should not make the raw version of blog widget post types discoverable. Clicking on the notification should take users directly to the formatted blog widget version. But also if somebody goes to the raw version, it would be nice to automatically redirect them. 

**Screenshots**

[Unformatted Raw Version](https://near.social/devgovgigs.near/widget/gigs-board.pages.Post?id=2029)
Screenshot 2023-11-22 at 11 23 51 AM

[Formatted Blog Widget Version](https://near.social/devgovgigs.near/widget/app?page=blog&id=2029)
Screenshot 2023-11-22 at 11 24 32 AM

**Desktop (please complete the following information):**
 - OS: OS
 - Browser: Safari
 - Version: Version 17.0 (19616.1.27.211.1)

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
Resolves #505

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@elliotBraem @ailisp 
After getting this merged I will ask Eugene to update [this](https://near.social/mob.near/widget/WidgetSource?src=mob.near/widget/Notification.Item.DevGov) to `devhub.near` account
and I have created a draft PR for near org [here](https://github.com/near/near-discovery-components/pull/544)
please let me know if any other change is required or if I m wrong somewhere
Thanks

File: src/DevGov/Notification/Item/Left.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const linkPropsQuery = Object.entries(linkProps)
+    .filter(([_key, nullable]) => (nullable ?? null) !== null)
+    .map(([key, value]) => `${key}=${value}`)
+    .join("&");
+
+  return `/#/${REPL_DEVHUB}/widget/devhub.page.${widgetName}${
+    linkPropsQuery ? "?" : ""
+  }${linkPropsQuery}`;
+}
+/* END_INCLUDE: "common.jsx" */
```
Comment (elliotBraem) [inline src/DevGov/Notification/Item/Left.jsx:18]: We can remove all of the above

Comment (elliotBraem) [reply src/DevGov/Notification/Item/Left.jsx:18]: Replace it with 
```
const { href } = VM.require("${REPL_DEVHUB}/widget/core.lib.url") || (() => {});
```

File: src/DevGov/Notification/Item/Left.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      ? "liked your"
+      : type == "reply"
+      ? "replied to your"
+      : type == "edit"
+      ? "edited your"
+      : type == "mention"
+      ? "mentioned you in their"
+      : "???"}{" "}
+    <a className="fw-bold text-muted" href={href("post", { id: props.post })}>
+      DevHub post
```
Comment (elliotBraem) [inline src/DevGov/Notification/Item/Left.jsx:37]: Then this <a tag should have 

```
href={to={href({
    widgetSrc: "${REPL_DEVHUB}/widget/app",
    params: {
      page: "post",
      id: props.id
    },
  })}
```

Although according to the story, the issue here is that for blogs, we should be pointing to blog instead:

```
to={href({
    widgetSrc: "${REPL_DEVHUB}/widget/app",
    params: {
      page: "blog",
    },
  })}
  ```
  
But we don't have access to the post type at this moment...

Maybe when we create the notification (author posting the blog), we could pass the postType of where it's coming from?

File: src/DevGov/Notification/Item/Right.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  const linkPropsQuery = Object.entries(linkProps)
+    .filter(([_key, nullable]) => (nullable ?? null) !== null)
+    .map(([key, value]) => `${key}=${value}`)
+    .join("&");
+
+  return `/#/${REPL_DEVHUB}/widget/devhub.page.${widgetName}${
+    linkPropsQuery ? "?" : ""
+  }${linkPropsQuery}`;
+}
+/* END_INCLUDE: "common.jsx" */
```
Comment (elliotBraem) [inline src/DevGov/Notification/Item/Right.jsx:18]: Remove above and replace with the VM require.

Need to fix the href below

**Review by elliotBraem — CHANGES_REQUESTED** [review]

**Comment by elliotBraem** [comment]
@Megha-Dev-19 it's a good start!

As per testing, this is something we could do:

For near.social, copy and paste the mob.near/widget/NotificationFeed code to local (and any relevant child components) so that we are serving the code locally. It could just be to a temp directory under src/ or something like that.

Then, we can create or edit a blog from [devhub-test](https://near.org/devhub.near/widget/app?page=community&handle=devhub-test) community, adding a mention to your name. We should be able to look at this new notification through our local "forked" notification feed.

**Comment by Megha-Dev-19** [comment]
@elliotBraem @ailisp can we merge this PR.? Will update the code further according to contract changes?
I have applied your suggestions

File: src/DevGov/Notification/Item/Left.jsx (hunk 2 above)
Comment (Megha-Dev-19) [reply src/DevGov/Notification/Item/Left.jsx:29]: yeah we don't know when it's a blog or post notification

**Review by ailisp — APPROVED** [review]
This looks good to me. @elliotBraem Could you review again? It's currently blocking from your request changes.

**Review by elliotBraem — APPROVED** [review]

**Pull request merged**